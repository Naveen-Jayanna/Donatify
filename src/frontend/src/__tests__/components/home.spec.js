import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../../components/home';

describe('homeComponent', () => {
	it('renders without crashing', () => {
		localStorage.setItem('userLogonDetails', JSON.stringify({ userId: 1, signInTime: new Date(), signInStatus: true }));
		const component = renderer.create(<Home />);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('initial state set correctly', () => {
		localStorage.setItem('userLogonDetails', JSON.stringify({}));
		const component = renderer.create(<Home />);
		const expectedOutput = {
			collapsed: true,
			content: '',
			userLogonDetails: {}
		};
		const actualOutput = component.getInstance().state;
		expect(expectedOutput).toEqual(actualOutput);
	});

	it('setCollapsed function updates state correctly', () => {
		localStorage.setItem('userLogonDetails', JSON.stringify({}));
		const component = renderer.create(<Home />);
		const expectedOutput = {
			collapsed: false,
			content: '',
			userLogonDetails: {}
		};
		// component.getInstance().setState = jest.fn();
		component.getInstance().setCollapsed();
		const actualOutput = component.getInstance().state;
		expect(expectedOutput).toEqual(actualOutput);
		// expect(component.getInstance().setState).toHaveBeenCalledWith(expectedOutput);
	});

	it('setContent function updates state correctly', () => {
		localStorage.setItem('userLogonDetails', JSON.stringify({}));
		const component = renderer.create(<Home />);
		const expectedOutput ={
			collapsed: true,
			content: 'PROFILE',
			userLogonDetails: {}
		};
		// component.getInstance().setState = jest.fn();
		component.getInstance().preventDefault = jest.fn();
		component.getInstance().setContent('PROFILE');
		const actualOutput = component.getInstance().state;
		expect(expectedOutput).toEqual(actualOutput);
	});
});
