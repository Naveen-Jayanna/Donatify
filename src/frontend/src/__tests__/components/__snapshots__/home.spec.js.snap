// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homeComponent renders without crashing 1`] = `
<div>
  <section
    className="ant-layout ant-layout-has-sider"
  >
    <aside
      className="ant-layout-sider ant-layout-sider-dark ant-layout-sider-collapsed"
      style={
        Object {
          "flex": "0 0 80px",
          "maxWidth": "80px",
          "minWidth": "80px",
          "width": "80px",
        }
      }
    >
      <div
        className="ant-layout-sider-children"
      >
        <div
          className="logo"
        />
        <ul
          className="ant-menu ant-menu-root ant-menu-vertical ant-menu-dark ant-menu-inline-collapsed"
          data-menu-list={true}
          onKeyDown={[Function]}
          role="menu"
          tabIndex={0}
        >
          <li
            className="ant-menu-item ant-menu-item-selected"
            data-menu-id="rc-menu-uuid-test-history"
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={Object {}}
            tabIndex={-1}
          >
            <span
              aria-label="history"
              className="anticon anticon-history ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="history"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M536.1 273H488c-4.4 0-8 3.6-8 8v275.3c0 2.6 1.2 5 3.3 6.5l165.3 120.7c3.6 2.6 8.6 1.9 11.2-1.7l28.6-39c2.7-3.7 1.9-8.7-1.7-11.2L544.1 528.5V281c0-4.4-3.6-8-8-8zm219.8 75.2l156.8 38.3c5 1.2 9.9-2.6 9.9-7.7l.8-161.5c0-6.7-7.7-10.5-12.9-6.3L752.9 334.1a8 8 0 003 14.1zm167.7 301.1l-56.7-19.5a8 8 0 00-10.1 4.8c-1.9 5.1-3.9 10.1-6 15.1-17.8 42.1-43.3 80-75.9 112.5a353 353 0 01-112.5 75.9 352.18 352.18 0 01-137.7 27.8c-47.8 0-94.1-9.3-137.7-27.8a353 353 0 01-112.5-75.9c-32.5-32.5-58-70.4-75.9-112.5A353.44 353.44 0 01171 512c0-47.8 9.3-94.2 27.8-137.8 17.8-42.1 43.3-80 75.9-112.5a353 353 0 01112.5-75.9C430.6 167.3 477 158 524.8 158s94.1 9.3 137.7 27.8A353 353 0 01775 261.7c10.2 10.3 19.8 21 28.6 32.3l59.8-46.8C784.7 146.6 662.2 81.9 524.6 82 285 82.1 92.6 276.7 95 516.4 97.4 751.9 288.9 942 524.8 942c185.5 0 343.5-117.6 403.7-282.3 1.5-4.2-.7-8.9-4.9-10.4z"
                />
              </svg>
            </span>
            <span
              className="ant-menu-title-content"
            >
              HISTORY
            </span>
          </li>
          <li
            className="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-donate"
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={Object {}}
            tabIndex={-1}
          >
            <span
              aria-label="heart"
              className="anticon anticon-heart ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="heart"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
                />
              </svg>
            </span>
            <span
              className="ant-menu-title-content"
            >
              DONATE
            </span>
          </li>
          <li
            className="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-profile"
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={Object {}}
            tabIndex={-1}
          >
            <span
              aria-label="user"
              className="anticon anticon-user ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="user"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                />
              </svg>
            </span>
            <span
              className="ant-menu-title-content"
            >
              PROFILE
            </span>
          </li>
          <li
            className="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-marketPlace"
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={Object {}}
            tabIndex={-1}
          >
            <span
              aria-label="shop"
              className="anticon anticon-shop ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="shop"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M882 272.1V144c0-17.7-14.3-32-32-32H174c-17.7 0-32 14.3-32 32v128.1c-16.7 1-30 14.9-30 31.9v131.7a177 177 0 0014.4 70.4c4.3 10.2 9.6 19.8 15.6 28.9v345c0 17.6 14.3 32 32 32h676c17.7 0 32-14.3 32-32V535a175 175 0 0015.6-28.9c9.5-22.3 14.4-46 14.4-70.4V304c0-17-13.3-30.9-30-31.9zM214 184h596v88H214v-88zm362 656.1H448V736h128v104.1zm234 0H640V704c0-17.7-14.3-32-32-32H416c-17.7 0-32 14.3-32 32v136.1H214V597.9c2.9 1.4 5.9 2.8 9 4 22.3 9.4 46 14.1 70.4 14.1s48-4.7 70.4-14.1c13.8-5.8 26.8-13.2 38.7-22.1.2-.1.4-.1.6 0a180.4 180.4 0 0038.7 22.1c22.3 9.4 46 14.1 70.4 14.1 24.4 0 48-4.7 70.4-14.1 13.8-5.8 26.8-13.2 38.7-22.1.2-.1.4-.1.6 0a180.4 180.4 0 0038.7 22.1c22.3 9.4 46 14.1 70.4 14.1 24.4 0 48-4.7 70.4-14.1 3-1.3 6-2.6 9-4v242.2zm30-404.4c0 59.8-49 108.3-109.3 108.3-40.8 0-76.4-22.1-95.2-54.9-2.9-5-8.1-8.1-13.9-8.1h-.6c-5.7 0-11 3.1-13.9 8.1A109.24 109.24 0 01512 544c-40.7 0-76.2-22-95-54.7-3-5.1-8.4-8.3-14.3-8.3s-11.4 3.2-14.3 8.3a109.63 109.63 0 01-95.1 54.7C233 544 184 495.5 184 435.7v-91.2c0-.3.2-.5.5-.5h655c.3 0 .5.2.5.5v91.2z"
                />
              </svg>
            </span>
            <span
              className="ant-menu-title-content"
            >
              MARKET PLACE
            </span>
          </li>
          <li
            className="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-logout"
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={Object {}}
            tabIndex={-1}
          >
            <span
              aria-label="logout"
              className="anticon anticon-logout ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="logout"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M868 732h-70.3c-4.8 0-9.3 2.1-12.3 5.8-7 8.5-14.5 16.7-22.4 24.5a353.84 353.84 0 01-112.7 75.9A352.8 352.8 0 01512.4 866c-47.9 0-94.3-9.4-137.9-27.8a353.84 353.84 0 01-112.7-75.9 353.28 353.28 0 01-76-112.5C167.3 606.2 158 559.9 158 512s9.4-94.2 27.8-137.8c17.8-42.1 43.4-80 76-112.5s70.5-58.1 112.7-75.9c43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 7.9 7.9 15.3 16.1 22.4 24.5 3 3.7 7.6 5.8 12.3 5.8H868c6.3 0 10.2-7 6.7-12.3C798 160.5 663.8 81.6 511.3 82 271.7 82.6 79.6 277.1 82 516.4 84.4 751.9 276.2 942 512.4 942c152.1 0 285.7-78.8 362.3-197.7 3.4-5.3-.4-12.3-6.7-12.3zm88.9-226.3L815 393.7c-5.3-4.2-13-.4-13 6.3v76H488c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h314v76c0 6.7 7.8 10.5 13 6.3l141.9-112a8 8 0 000-12.6z"
                />
              </svg>
            </span>
            <span
              className="ant-menu-title-content"
            >
              LOGOUT
            </span>
          </li>
        </ul>
        <div
          aria-hidden={true}
          style={
            Object {
              "display": "none",
            }
          }
        />
      </div>
    </aside>
    <section
      className="ant-layout site-layout"
    >
      <header
        className="ant-layout-header site-layout-background"
        style={
          Object {
            "padding": 0,
          }
        }
      >
        <span
          aria-label="menu-unfold"
          className="anticon anticon-menu-unfold trigger"
          onClick={[Function]}
          role="img"
          tabIndex={-1}
        >
          <svg
            aria-hidden="true"
            data-icon="menu-unfold"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM142.4 642.1L298.7 519a8.84 8.84 0 000-13.9L142.4 381.9c-5.8-4.6-14.4-.5-14.4 6.9v246.3a8.9 8.9 0 0014.4 7z"
            />
          </svg>
        </span>
      </header>
      <main
        className="ant-layout-content site-layout-background"
        style={
          Object {
            "margin": "24px 16px",
            "minHeight": 280,
            "padding": 24,
          }
        }
      />
    </section>
  </section>
</div>
`;
